### PR TITLE
BoxDecoration's borderRadius should be able to handle a value of double.INFINITY

### LINKED ISSUE #939: BoxDecoration's borderRadius should be able to handle a value of double.INIFINITY
Currently it appears to ignore this value as if it's 0.0

### PR DESCRIPTION
**Pull request by abarth**
We just need to enforce some finite clamp and Skia will do the rest.

Fixes #939

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: sky/packages/sky/lib/painting/box_painter.dart [hunk 1]
```diff
@@ -474,6 +474,14 @@ class BoxPainter {
     return hasUniformWidth;
   }
 
+  double _getEffectiveBorderRadius(Rect rect) {
+    double shortestSide = rect.shortestSide;
+    // In principle, we should use shortestSide / 2.0, but we don't want to
+    // run into floating point rounding errors. Instead, we just use
+    // shortestSide and let sky.Canvas do any remaning clamping.
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/box_painter.dart]: typo in remaining

**Comment by Hixie** [comment]
LGTM

File: sky/packages/sky/lib/painting/box_painter.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/box_painter.dart]: Fixed.

**Pull request merged**